src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -261,13 +266,73 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
             tr.style.height = getComputedStyle(tr2).height;
         });
 
-        // Adjust fixed columns data to data height
-        const contentTd = r1c1.querySelector('tr > td:first-of-type');
-        if (contentTd) {
-            const contentTr = contentTd.parentElement as HTMLElement;
-
-            this.stylesheet.setRule('.dash-fixed-column tr', `height: ${getComputedStyle(contentTr).height};`);
-        }
+        if (fixed_columns) {
+            const lastVisibleTd = r1c0.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
+
+            let it = 0;
+            let currentWidth = r1c0.getBoundingClientRect().width;
+            let lastWidth = currentWidth;
+
+            do {
+                lastWidth = currentWidth
+
+                // Force first column containers width to match visible portion of table
+                if (lastVisibleTd) {
+                    const r1c0FragmentBounds = r1c0.getBoundingClientRect();
+                    const lastTdBounds = lastVisibleTd.getBoundingClientRect();
+                    currentWidth = lastTdBounds.right - r1c0FragmentBounds.left;
+
+                    const width = `${currentWidth}px`;
+
+                    r0c0.style.width = width;
+                    r1c0.style.width = width;
+                }
+
+                // Force second column containers width to match visible portion of table
+                const firstVisibleTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns + 1})`);
+                if (firstVisibleTd) {
+                    const r1c1FragmentBounds = r1c1.getBoundingClientRect();
+                    const firstTdBounds = firstVisibleTd.getBoundingClientRect();
+
+                    const width2 = firstTdBounds.left - r1c1FragmentBounds.left;
+                    r0c1.style.marginLeft = `-${width2}`;
+                    r0c1.style.marginRight = `${width2}`;
+                    r1c1.style.marginLeft = `-${width2}`;
+                    r1c1.style.marginRight = `${width2}`;
```

Comment:
If there are fixed columns, align the cell's table in such a way as to make the first visible column left-aligned with the cell.